js/ui/html_editor/utils/templates_storage.js

```diff
@@ -0,0 +1,39 @@
+import { isDefined } from '../../../core/utils/type';
+
+export default class TemplatesStorage {
+    constructor() {
+        this._widgetsStorage = {};
+    }
+
+    set({ widgetKey, marker }, value) {
+        let widgetTemplates = this._widgetsStorage[widgetKey];
+        if(!widgetTemplates) {
+            this._widgetsStorage[widgetKey] = widgetTemplates = widgetTemplates = {};
+        }
+
+        widgetTemplates[marker] = value;
+    }
+
+    get({ widgetKey, marker }) {
+        if(isDefined(widgetKey)) {
+            const widgetTemplates = this._widgetsStorage[widgetKey];
+            return widgetTemplates ? widgetTemplates[marker] : undefined;
+        }
+
+        const widgetsStorageKeys = Object.keys(this._widgetsStorage);
+
+        return widgetsStorageKeys.length > 0 ? this._widgetsStorage[widgetsStorageKeys[widgetsStorageKeys.length - 1]][marker] : undefined;
+    }
+
+    delete({ widgetKey, marker }) {
+        const widgetTemplates = this._widgetsStorage[widgetKey];
```

Comment:
I think this function can be implement just like
```javascript
    delete this._storage[editorKey][marker];
```
Why do we need remove all the entry if it's empty? Is it a real scenario?

Reply:
if do not do this, we will have empty object in this._storage[editorKey]